.github/copilot-instructions.md

```diff
@@ -0,0 +1,1399 @@
+## Project Overview
+
+### What is cdxgen?  
+`cdxgen` is a universal, polyglot CLI tool, library, REPL, and server for generating a compliant CycloneDX Bill of Materials (BOM). It scans source code, dependencies, container images, and more, across many languages and ecosystems — and outputs SBOMs in the standard CycloneDX format.  
+:contentReference[oaicite:2]{index=2}
+
+### Why it exists  
+Modern software systems are composed of many third-party and open-source components. Without visibility into all the components your software uses, you cannot accurately assess risk, vulnerabilities, licensing issues or supply chain exposure. Traditional “manifest-only” SBOM tools often fall short in enterprise environments.  
+Hence `cdxgen` was designed to be **precise**, **explainable**, and **comprehensive**:  
+- It aims for *explainability* — not just listing components, but providing evidence. :contentReference[oaicite:3]{index=3}  
+- It supports a wide range of technologies and use-cases (source code, containers, VMs).  
+- It is built to integrate into CI/CD pipelines and large scale security workflows.
+
+### Core Capabilities  
+- Generate SBOMs (Software Bill of Materials) for many project types and languages. :contentReference[oaicite:4]{index=4}  
+- Support for container image analysis, and even live systems / VMs in some cases. :contentReference[oaicite:5]{index=5}  
+- Output formats include JSON, XML and other formats per the CycloneDX specification versions 1.4–1.6. :contentReference[oaicite:6]{index=6}  
+- Designed for usage by developers, auditors, security researchers, and DevSecOps teams.
+
+### Ideal Use-Cases  
+- In a CI/CD pipeline, automatically generate a BOM for every build and upload it to a tracking system (e.g., Dependency‑Track) for vulnerability/license monitoring.  
+- For compliance or regulatory reporting, produce an audit-ready list of all components with metadata, licenses, and potential vulnerabilities.  
+- For container/image security, extract BOMs from images and VMs to feed into risk assessment workflows.
+
+### Technology & Ecosystem  
+- Written in Node.js / TypeScript.  
```

Comment:
JavaScript and not TypeScript. cdxgen-plugins-bin is not an architecture, but merely a library to support certain languages and binary types.